crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -685,7 +685,12 @@ impl<'db> TypeInferenceBuilder<'db> {
                     let symbol = symbols.symbol_by_name(id).unwrap();
                     if !symbol.is_defined() || !self.scope.is_function_like(self.db) {
                         // implicit global
-                        Some(module_global_symbol_ty_by_name(self.db, self.file, id))
+                        let mut unbound_ty =
+                            module_global_symbol_ty_by_name(self.db, self.file, id);
```

Comment:
This is a pre-existing issue, so it could be done in a separate PR, but I think we also need to ensure we don't make _this_ fallback if we are already inferring types in the module global scope.